bravado/http_future.py

```diff
@@ -101,18 +109,58 @@ def __init__(self, future, response_adapter, operation=None,
         self.response_callbacks = response_callbacks or REQUEST_OPTIONS_DEFAULTS['response_callbacks']
         self.also_return_response = also_return_response
 
-    @reraise_errors
-    def result(self, timeout=None):
+    def response(self, timeout=None, fallback_response=None, exceptions_to_catch=FALLBACK_EXCEPTIONS):
         """Blocking call to wait for the HTTP response.
 
+        :param timeout: Number of seconds to wait for a response. Defaults to
+            None which means wait indefinitely.
+        :type timeout: float
+        :param fallback_response: callable that accepts an exception as argument and returns the
+            swagger result to use in case of errors
+        :type fallback_response: callable that takes an exception and returns a fallback swagger result
+        :param exceptions_to_catch: Exception classes to catch and call `fallback_response`
+            with. Has no effect if `fallback_response` is not provided. By default, `fallback_response`
+            will be called for read timeout and server errors (HTTP 5XX).
+        :type exceptions_to_catch: List/Tuple of Exception classes.
+        :return: A BravadoResponse instance containing the swagger result and response metadata.
+        """
+        swagger_result = None
+        try:
+            incoming_response = self._get_incoming_response(timeout)
+            if self.operation is not None:
+                unmarshal_response(
+                    incoming_response,
+                    self.operation,
+                    self.response_callbacks,
+                )
+
+                swagger_result = incoming_response.swagger_result
+            elif incoming_response.status_code >= 300:
+                raise make_http_exception(response=incoming_response)
+        except exceptions_to_catch as e:
+            if fallback_response:
+                swagger_result = fallback_response(e)
+                incoming_response = None
+            else:
+                raise
+
+        return BravadoResponse(
+            result=swagger_result,
+            incoming_response=incoming_response,
+        )
+
+    def result(self, timeout=None):
+        """DEPRECATED: please use the `response()` method instead.
```

Comment:
Do we really want to deprecate this?
Personally I think that result is useful to get the swagger result object, while the new method (``response``) if useful to get also meta-data, etc.

Reply:
But `response` can be used to get the swagger result object, no? I think it helps in avoiding confusion, and standardizing on one interface. Why would you not want to deprecate it? If/when we actually remove the code is a different question.

Reply:
agreed

Reply:
I think we should keep `.result` around, otherwise we'll block people from upgrading. Our goal was to stay backwards-compatible no?

Reply:
Yep we're keeping it, just advising people to use `response()`.